layout/post.pug

```diff
@@ -57,10 +57,30 @@ block content
                         data-comment_valine_key= theme.comment_valine.appKey
                     )
                 if theme.comment_disqus && theme.comment_disqus.enable
-                    div#disqus_thread
                     article.mt-6.comment-container#disqus
-                        script="var disqus_config = function () {this.page.url = '"+theme.comment_disqus.url+url_for(page.path)+"';this.page.identifier = '"+theme.comment_disqus.name+"';};"
+                        div#disqus_thread
+                        script="var disqus_config = function () {this.page.url = window.location.href;this.page.identifier = document.title;};"
                         script="(function() {var d = document, s = d.createElement('script');s.src = 'https://"+theme.comment_disqus.name+".disqus.com/embed.js';s.setAttribute('data-timestamp', +new Date());(d.head || d.body).appendChild(s);})();"
                         script(id="dsq-count-scr", src="//blog-pubgj2togw.disqus.com/count.js", async)
+                script.
+                        String.prototype.replaceAll = function (FindText, RepText) {
+                          return this.replace(new RegExp(FindText, "g"), RepText);
+                        }
+                        function html_encode(str) 
+                        { 
+                            var s = ""; 
+                            if (str.length == 0) return ""; 
+                            s = str.replace(/</g, "&lt;");  
+                            s = s.replace(/>/g, "&gt;");
+                            return s; 
+                        } 
+                        document.addEventListener('DOMContentLoaded', (event) => {
+                          document.querySelectorAll('pre code').forEach((block) => {
+                            block.innerHTML = html_encode(block.innerHTML);
+                            hljs.highlightBlock(block);
+                          });
+                        });
 block script
+    if theme.highlight && theme.highlight.enable && hljs
```

Comment:
your original code in post.js seem donot consider the theme.highlight.enable, so i move it to here
**And this may be wrong in some ways, if is, plz fix this**

Reply:
and u will see that i am change the highlight.pack.js, that is because it is out of date, the newest version is v10.7.1 and the file in this project is 10.4.1, so i upgrade it